rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/dataspace/RestDataspaceImpl.java

```diff
@@ -302,18 +319,19 @@ public Response create(@HeaderParam("sessionid") String sessionId,
             logger.error(String.format("Cannot create folder for %s in %s", pathname, dataspacePath), e);
             throw rethrow(e);
         } catch (Throwable e) {
-            logger.error(
-                    String.format(
-                            "Cannot retrieve metadata for %s in %s.",
-                            pathname, dataspacePath), e);
+            logger.error(String.format("Cannot create folder for %s in %s", pathname, dataspacePath), e);
             throw rethrow(e);
         }
     }
 
-    private Response componentResponse(String type, FileObject fo) throws FileSystemException {
-        return ("list".equals(type)) ? Response.ok(FileSystem.list(fo), MediaType.APPLICATION_JSON).build()
-                : Response.status(Response.Status.BAD_REQUEST).entity(
+    private Response componentResponse(String type, FileObject fo, List<String> includes, List<String> excludes) throws FileSystemException {
+        switch (type) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1301)